packages/ipns/src/index.ts

```diff
@@ -167,15 +167,22 @@ export interface DNSResolver {
   (domain: string, options?: ResolveDnsLinkOptions): Promise<string>
 }
 
-export interface ResolveDNSOptions extends AbortOptions, ProgressOptions<ResolveDnsLinkProgressEvents> {
+export interface ResolveDNSOptions extends AbortOptions, ProgressOptions<ResolveProgressEvents | IPNSRoutingEvents | ResolveDnsLinkProgressEvents> {
```

Comment:
`resolveDns` (which takes `ResolveDNSOptions`) ultimately calls `#resolve`, which is also called by `resolve` (which takes `ResolveOptions`). 

This could be cleaned up but I'll leave this for @achingbrain to decide if he wants this cleaned up further. Ultimately, `#resolve` should take a superset of `resolve` and `resolveDns` option types.